spec/lib/common/connection-spec.js

```diff
@@ -5,113 +6,254 @@
+                            'exchanges':
+                                'on.test'
+                        })
+                    })
+                    this.subject.start().then(function () {
+
+                        self.subject.exchanges.should.not.be.undefined;
+                    });
                 });
             });
-        });
 
-        describe('exchanges', function () {
-            it('should be undefined if not connected', function () {
-                expect(this.subject.exchanges).equal(undefined);
-            });
+            describe('exchange', function ()  {
+                it('should not provide an exchange if connection not established ', function () {
+                    var self = this;
+                    return this.subject.exchange()
+                        .catch(function (Error) {
+                            console.log(Error);
+                        });
+                });
 
-            it('should have exchanges if connected', function () {
-                var self = this;
+                it('should provide an exchange if connection established and options is a valid object', function () {
+                    var self = this;
+                    amqp.createConnection = sinon.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                callback(null, {});
+                            },
+                            'exchange': function (a, b, callback){
+                                callback('Hello', {})
+                            },
+                            'exchanges':
+                                'on.test'
+                        })
+                    })
+                    this.subject.start().then(function () {
+                        if (self.subject.connected == true) {
+                            var name = 'amqp://localhost';
+                            return self.subject.exchange(name, {url: 'amqp://localhost'}, function(callback) {
+                                callback('exchange', {});
+                            });
+                        }
+                    });
+                });
+
+                it('should error out if objects is not valid ', function () {
+                    var self = this;
+                    amqp.createConnection = sinon.spy(function () {
+                        return ({
+                            'on': function (a, callback) {
+                                callback(null, {});
+                            },
+                            'exchange': function (a, b, callback){
+                                callback('Hello', {})
+                            },
+                            'exchanges':
+                                'on.test'
+                        })
+                    })
+                    this.subject.start().then(function () {
+                        if (self.subject.connected == true) {
```

Comment:
Do we want a failure case if `self.subject.connected` does not equal true? Otherwise it's a noop test in this case.